docs/proposals/20200804-windows-support.md

```diff
@@ -0,0 +1,271 @@
+The infrastructure provider is responsible for putting the cloud-init script in the right location. 
+When the VM is booted, the cloud-init script runs automatically. 
+
+Cloud-init does not have Windows support. An alternative product is [cloudbase-init](https://github.com/cloudbase/cloudbase-init). 
+Cloudbase-init functions in the same way as cloud-init and can consume cloud-init scripts as provided by the Cluster API Bootstrap Provider Kubeadm.  
+By using cloudbase-init, Windows can leverage the existing solutions and stay up to date with the latest changes in CABPK.
+
+#### VHD Creation
+Using cloudbase-init requires the creation of a VHD with the tooling installed on it since it is not 
+provided out of the box by any cloud providers.  To address this we will provide packer scripts as part of 
+the [image-builder project](https://github.com/kubernetes-sigs/image-builder) that will pre-install 
+cloudbase init.  It is important to note that while scripts can be provided to build a  VHD.  All VHD’s 
+built will need to adhere to [Windows licensing requirements](https://www.microsoft.com/en-us/licensing/product-licensing/windows-server).
+
+AKS-Engine has an example implementation for using packer and scripts to do VHD configuration: https://github.com/Azure/aks-engine/blob/master/vhd/packer/windows-vhd-builder.json
+
+Although the Linux implementation in image-builder uses Ansible for configuration, Windows will not share 
+the configuration. Therefore, Ansible is not a requirement for Windows image builder implementation.  
+The main requirement for VHD building is that the build can be started from a Linux or Windows machine.
+
+#### Kubelet and other component configuration
+Due to the lack of privileged containers in Windows, a combination of `PreKubeadmCommands`/`PostKubeadmCommands` 
+scripts and wins.exe can be used to configure the nodes. Wins.exe is currently provided as a way to  bootstrap nodes along with kubeadm in the [Kubernetes documentation for adding Windows nodes](https://kubernetes.io/docs/tasks/administer-cluster/kubeadm/adding-windows-nodes/).  
+The components from the [preparenode script](https://kubernetes.io/docs/tasks/administer-cluster/kubeadm/adding-windows-nodes/#joining-a-windows-worker-node) can be used during VHD creation.
+
+In the future with the introduction of the [Privileged Containers for Windows containers](https://docs.google.com/document/d/12EUtMdWFxhTCfFrqhlBGWV70MkZZPOgxw0X-LTR0VAo/edit#) we can use Privileged containers in place of wins.exe enabled containers. 
+As the privileged container support is still being discussed, the replacement of the `PreKubeadmCommands`/`PostKubeadmCommands` kubeadm scripts might be possible but will need to be addressed at a later point in time. 
+
+The plan during Alpha implementation is that each of the infrastructure providers will provide their own `PreKubeadmCommands`/`PostKubeadmCommands` scripts. 
```

Comment:
Do we expect these to be very different? why not go for a common script right away?

Reply:
Depending on components such as runtime/cni, different settings/configurations might be required.  I don't have a clear enough picture across the infrastructure providers to make a determination currently and didn't want to hold back alpha support.  

A script that will be useful is a [windows automatic retry based on CAPH](https://github.com/microsoft/cluster-api-provider-azurestackhci/blob/e3fc20c8d835a2b854a1c05ade2f8bf91ea4db53/templates/flavors/base/cluster-template.yaml#L65) and I will update the CEAP to add that as alpha feature that should be added.